### PR TITLE
Add possibility to run inference benchmarks on XPU device

### PR DESCRIPTION
**Pull request by DamianSzwichtenberg**
Add possibility to run inference benchmarks on XPU device.
Exemplary CMD:
`python inference_benchmark.py --device xpu --datasets Reddit --models sage --eval-batch-sizes 1024 --num-layers 2 --num-hidden-channels 64 --profile`

### CODE REVIEW

File: benchmark/inference/inference_benchmark.py [hunk 1]
```diff
@@ -256,6 +281,8 @@ def run(args: argparse.ArgumentParser):
     argparser = argparse.ArgumentParser('GNN inference benchmark')
     add = argparser.add_argument
 
+    add('--device', choices=['cpu', 'cuda', 'xpu'], default='cuda',
```
Comment (kgajdamo) [inline benchmark/inference/inference_benchmark.py:284]:
```suggestion
    add('--device', choices=['cpu', 'cuda', 'xpu'], default='cpu',
```

File: benchmark/utils/utils.py [hunk 2]
```diff
@@ -194,3 +194,12 @@ def test(model, loader, device, hetero, progress_bar=True,
             total_examples += batch_size
             total_correct += int((pred == batch.y[:batch_size]).sum())
     return total_correct / total_examples
+
+
+@contextmanager
+def xpu_profiler(export_chrome_trace=True):
```
Comment (kgajdamo) [inline benchmark/utils/utils.py:200]: An analogous pytorch profiler function is located in the [torch_geometric/profile/profile.py](https://github.com/pyg-team/pytorch_geometric/blob/376786f91f691eb2e98dca263029dac4a2963808/torch_geometric/profile/profile.py). I wonder if the one for xpu should also be there.

Comment (DamianSzwichtenberg) [reply benchmark/utils/utils.py]: Good idea 😉 Done.

**Review by kgajdamo — APPROVED** [review]
Looks good to me.

**Comment by JakubPietrakIntel** [comment]
LGTM!

**Review by JakubPietrakIntel — APPROVED** [review]
LGTM!

**Pull request merged**